proto/docs/working-with-protobufs.md

```diff
@@ -0,0 +1,63 @@
+# Working with Protobufs
```

Comment:
I think it would be helpful to briefly answer "Why we chose Protobufs" in this section, and describe how they are used in our architecture from instrumentation to ingestion.

Reply:
This section was added @ac360